pynicotine/gtkgui/userbrowse.py

```diff
@@ -688,34 +711,336 @@ def on_select_dir(self, selection):
+                if self.user == config.sections["server"]["login"]:
+                    if (prs_ctrl) or self.num_selected_files <= 0:
+                        # Ctrl+Alt+Enter: Open if File Manager
+                        self.on_file_manager()
+                    else:
+                        # Alt+Enter: File Properties
+                        self.on_file_properties()
+
+                elif (prs_ctrl) or self.num_selected_files <= 0:
+                    # Alternate Download Folder Into... Location
+                    self.on_download_directory_to()
+
+                else:
+                    self.on_file_properties()
+
+                return True
+
+        self.on_key_press_event_browse(*args)
         return False
 
-    def on_file_key_press_event(self, *args):
+    def on_key_press_event_browse(self, *args):
 
         keyval, keycode, state, widget = get_key_press_event_args(*args)
-        keycodes, mods = parse_accelerator("<Alt>Return")
 
-        if state & mods and keycode in keycodes:
-            self.on_file_properties()
+        _keys, mods_primary = parse_accelerator("<Primary>")
+        _keys, mods_shift = parse_accelerator("<Shift>")
+        _keys, mods_alt = parse_accelerator("<Alt>")
+
+        prs_none = prs_ctrl = prs_shift = prs_alt = False
+
+        if state & mods_primary:  # <GDK_CONTROL_MASK>
+            prs_ctrl = True
+
+        if state & mods_shift:  # <GDK_SHIFT_MASK>
+            prs_shift = True
+
+        if state & mods_alt:  # <GDK_MOD1_MASK>
+            prs_alt = True
+
+        if prs_ctrl == prs_shift == prs_alt is False:
+            prs_none = True  # <GDK_MOD2_MASK> (plain)
+
+        """ Key Bindings (this UserBrowses Page) """
+
+        # Ctrl+F: Focus Find TextEntry
+        keycodes, mods = parse_accelerator("<Primary>f")
+
+        if state & mods and keycode in keycodes and (not prs_shift and not prs_alt):
+            self.SearchEntry.grab_focus()
             return True
 
+        # Ctrl+G or F3: Find Next Match
+        keycodes_g, _mods_g = parse_accelerator("<Primary>g")
+        keycodes_f3, _mods_f3 = parse_accelerator("F3")
+
+        if (keycode in keycodes_g and prs_ctrl) or (keycode in keycodes_f3 and not prs_ctrl):
+            if (not prs_alt):  # Avoid any conflict
+                if (not prs_shift) and self.SearchEntry.get_text() != "":
+                    self.on_search()
+
+                elif prs_shift:
+                    # Shift+Ctrl+G or Shift+F3: Find Previous Match
+                    # ToDo: {self.search_position = self.search_position - 2}
+                    self.on_search()
+
+                else:
+                    self.SearchEntry.grab_focus()
+
+                return True
+
+        # Ctrl+R or F5: Refresh
+        keycodes_r, _mods_r = parse_accelerator("<Primary>r")
+        keycodes_f5, _mods_f5 = parse_accelerator("F5")
+
+        if (keycode in keycodes_r and prs_ctrl and not prs_shift) or (keycode in keycodes_f5 and prs_none):
```

Comment:
The 'mods' returned by Gtk.accelerator_parse is woefully inadequate for our purposes, because it validates even when additional modifier keys are also pressed (eg: Ctrl+R is not the same as Shift+Ctrl+R). This is where key conflicts arise.

For now, I have done this hack that is currently needed to get the shortcuts working properly, but I feel there must be a cleaner way of doing this. Perhaps our parse_accelerator() function can be expanded to better handle multiple keys, or maybe it would be safer to create a new function such as parse_shortcut() that is more suited to our needs.